pennylane/templates/state_preparations/state_prep_mps.py

```diff
@@ -188,6 +256,66 @@ def _primitive_bind_call(cls, mps, wires, id=None):
             return type.__call__(cls, mps=mps, wires=wires, id=id)  # pragma: no cover
         return cls._primitive.bind(*mps, wires=wires, id=id)
 
+    def decomposition(self):  # pylint: disable=arguments-differ
+        filtered_hyperparameters = {
+            key: value for key, value in self.hyperparameters.items() if key != "input_wires"
+        }
+        return self.compute_decomposition(
+            self.parameters, wires=self.hyperparameters["input_wires"], **filtered_hyperparameters
+        )
+
+    @staticmethod
+    def compute_decomposition(mps, wires, work_wires):  # pylint: disable=arguments-differ
+        r"""Representation of the operator as a product of other operators.
+
+        Args:
+            mps (List[Array]):  list of arrays of rank-3 and rank-2 tensors representing an MPS state as a
+                product of site matrices. See the usage details section for more information.
+
+            wires (Sequence[int]): wires that the template acts on
+            work_wires (Sequence[int]): list of extra qubits needed. The bond dimension of the mps
+                is defined as ``2^len(work_wires)``
+
+        Returns:
+            list[.Operator]: Decomposition of the operator
+        """
+
+        if work_wires is None:
+            raise AssertionError("To decompose MPSPrep you must specify `work_wires`.")
```

Comment:
```suggestion
            raise AssertionError("To decompose qml.MPSPrep you must specify `work_wires`.")
```
Also we should update this to include the MPS used cannot have the bond dimension greater than `2^#work_wires`